src/Entities/Categories.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+Class Categories{
+    private int $id;
+    private string $name;
+    private int $total_stock;
+
+    public function displayByCategory(){
+        echo '<h3 class="text-2xl">' . $this->name . '</h3>';
+        echo '<span class="bg-teal-500 text-2xl px-2 py-1 rounded">' . $this->total_stock . '</span>';
+    }
```

Comment:
This method needs to be unit tested, however because your entity is created using PDO::FETCH_CLASS, you cannot instantiate this entity in order to test it.

What you should do instead is create a new class (like a category display service) that has the displayCategory method which accepts a CategoryEntity as an argument. This method could be static like your factory connect method.

That then makes this display code unit testable, and better follows the single responsibility principle as the entity strictly contains the data and doesn't do anything else.

Note: You will need to make this method return a string rather than echoing it - you cannot unit test something that echos